### ISSUE TITLE
Implement local deck saving as a CSV file

### ISSUE THREAD

**Issue by luzhanov** [luzhanov/anki-adder-fork#10]
PREREQUISITES:
- work in branch https://github.com/luzhanov/anki-adder-plus/tree/2.0.0_local_storage
- saving method is in form.js:137 -> $(".exportcardbutton").click
- save few cards to the local deck (e.g. "Deck A") using "Add to DB" button

ACCEPTANCE CRITERIA
- When user click on the orange button "Export to file" all data from the active deck (in our case - from the "Deck A") should be saved as a text file.
- Card should be loaded from the existing LDB record (take a look on existing code)
- File should be ready for the Anki importing - i think better is to use Tab separation
- All card's fields should be exported. I'm not sure about tags - please check is it possible to import them from CSV.
- After export localDB entry for the current deck should be dropped.

**Comment by luzhanov** [comment]
By the way, I noticed this package https://github.com/vvscode/js--anki-apkg-export

It would be great to use it in future, but it require a lot of refactoring for the current plugin... I think that it's better to start with quick & simple CSV solution and release it ASAP, and later think about implementing APKG export.

**Comment by vvscode** [comment]
I just try to export test deck from anki desktop -- looks like export file doesn't support tags ( at least text format ). But threre is a number of addons like `anki-crowd` which allow use json format with additional fields

**Comment by vvscode** [comment]
@luzhanov 

there are list of questions: 

1) Use current implementation ( via `click on link` ) or file storage api ( it need additional permissions )

2) about drop db -- anki support avoiding card duplication on import 
![https://www.dropbox.com/s/mh88972r7xgagcn/%D0%A1%D0%BA%D1%80%D0%B8%D0%BD%D1%88%D0%BE%D1%82%202017-01-23%2011.47.49.jpg?dl=1](https://www.dropbox.com/s/mh88972r7xgagcn/%D0%A1%D0%BA%D1%80%D0%B8%D0%BD%D1%88%D0%BE%D1%82%202017-01-23%2011.47.49.jpg?dl=1)

So either we just skip this functionality or mark exported records in db via special flag to avoid loosing user data

**Comment by luzhanov** [comment]
@vvscode 
1. What will be the difference of using file storage API approach?
2. Good point. First of all I'm afraid of using all free space in the localStorage if we will keep all data.
So I like how you implemented it right now - user will have a choice what to do.
Also I've added a check here ankiweb.js:131 to be sure that user will see if all space will be used.

Thank you for you changes!
Few small notes to finish this task:

1. Use .csv extension instead of .txt
2. Add support for the cards with any number of fields (while saving you can select from two-fields & three fields card models)
3. Move messages to the messages.json file and get it via chrome.i18n.getMessage()

**Comment by vvscode** [comment]
> What will be the difference of using file storage API approach?
 As far as I understand
1.1) we can ask filename / file path only once. And later just update existing file -- not sure that it's usefull 
1.2) we can be sure that data was saved to drop db ( in current implementation we can't be sure )


> I'm afraid of using all free space in the localStorage
Default limit for Chrome is 10 mb ( I suppose it's huge space for text only cards ). But if current case is ok - fine. 

> Use .csv extension instead of .txt
I use `.txt` cause anki desktop export decks into it. So just confirm that we need change file ext. and I'll do it

**Comment by luzhanov** [comment]
> 1.1) we can ask filename / file path only once. And later just update existing file -- not sure that it's usefull

I think it can be confusing for the user, so let's skip this.

> 1.2) we can be sure that data was saved to drop db ( in current implementation we can't be sure )

It's a good point. I propose to do the next: I will create a new issue for this and we can implement it later. For a moment it's not a stopper. What do you think?

> I use .txt cause anki desktop export decks into it. So just confirm that we need change file ext. and I'll do it

You're right - Anki can import/export both .txt and .csv, but I'm thinking about case if user will need some editing using Excel - in this case it can be opened in one-click.
Taking this into account I think that .csv it's better.

**Comment by vvscode** [comment]
https://github.com/luzhanov/anki-adder-plus/pull/13

**Comment by luzhanov** [comment]
Very cool! :) Thank you!
